### PR TITLE
KAFKA-527 FEAT Allow overriding of producer creation.

### PR DESCRIPTION
**Pull request by milanspre**

### CODE REVIEW

File: kafka/src/main/java/io/micronaut/configuration/kafka/KafkaProducerFactory.java [hunk 1]
```diff
@@ -130,6 +124,21 @@ public <K, V> Producer<K, V> getProducer(
...
+     * Creates kafka producer, could be overridden for further control.
+     *
+     * @param config properties for producer
+     * @param ks key serializer
+     * @param vs value serializer
+     * @param <K> key type
+     * @param <V> value type
+     * @return new instance of producer
+     */
+    protected <K, V> Producer<K, V> createProducer(Properties config, Serializer<K> ks, Serializer<V> vs) {
```
Comment (graemerocher) [inline kafka/src/main/java/io/micronaut/configuration/kafka/KafkaProducerFactory.java:138]: Add nullability annotations and `@since` declaration

**Review by graemerocher — CHANGES_REQUESTED** [review]

**Comment by dstepanov** [comment]
IMHO I would rename `KafkaProducerFactory` to `KafkaProducerRegistry` because it implements `ProducerRegistry` and extract this method to a new interface `KafkaProducerFactory` with a default impl with `@Requires(missing`. Users then can provide a custom implementation.

File: kafka/src/main/java/io/micronaut/configuration/kafka/KafkaProducerFactory.java (hunk 1 above)
Comment (milanspre) [reply kafka/src/main/java/io/micronaut/configuration/kafka/KafkaProducerFactory.java]: Added.

**Comment by graemerocher** [comment]
Replaced by https://github.com/micronaut-projects/micronaut-kafka/pull/619

**Pull request closed without merging**